src/main/java/teammates/common/datatransfer/SearchResultBundle.java

```diff
@@ -1,14 +1,9 @@
 package teammates.common.datatransfer;
 
 /**
- * The basic search result bundle object.
+ * The base class for search result bundle.
  */
-public class SearchResultBundle {
-
-    public int numberOfResults;
-
-    protected SearchResultBundle() {
-        // prevents instantiation; to be instantiated as children classes
-    }
+abstract class SearchResultBundle {
 
+    public abstract int size();
```

Comment:
I don't think even this is needed. All usages are equivalent to the size of a list, which is a public variable. In addition, the only usage of this method is in tests.